### PR TITLE
Close nav menu after clicking link

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Make sure to close the nav menu after clicking nested links.

Use router events to close the nav menu after clicking a link.

## Justify why they are needed

This should work out of the box but Next.js does something with the event propagation that makes it not work. This is a workaround. Read more here: https://github.com/radix-ui/primitives/issues/1301

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1577** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1577" target="_blank">Graphite</a> 
    * **PR #1585** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1585" target="_blank">Graphite</a>  👈
    * **PR #1579** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1579" target="_blank">Graphite</a> 
      * **PR #1581** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1581" target="_blank">Graphite</a> 
        * **PR #1582** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1582" target="_blank">Graphite</a> 
          * **PR #1583** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1583" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1585?utm_source=stack-comment).

**Review by alebedev — APPROVED** [review]
Looks good

File: apps/store/src/components/Header/TopMenuDesktop/TopMenuDesktop.tsx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import styled from '@emotion/styled'
-import React from 'react'
+import { useRouter } from 'next/router'
+import React, { useEffect } from 'react'
```
[the remark this answers is not in the document]
Comment (robinandeer) [reply apps/store/src/components/Header/TopMenuDesktop/TopMenuDesktop.tsx:3]: yeah not really needed in tsx files I guess

**Comment by robinandeer** [comment]
⏳ This pull request is set to merge as part of a Graphite merge job
Job ID: fCyIicIm0zvN6r4Dun6w.
See details on [graphite.dev](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1585)

**Pull request merged**